netsim/extra/repeater/plugin.py

```diff
@@ -0,0 +1,83 @@
+    return intf
+  
+  ifs = []
+  if nodename in [ i.node for i in link_data.interfaces ]:
+     link_data.pop('_linkname',None)
+     atts = [ i for i in link_data.interfaces if i.node==nodename ][0]      # Copy any interface atts
+     for c,clone in enumerate(clones):
+       l = data.get_box(link_data)
+       l.interfaces = [ atts+{ 'node': clone } ] + [ update_ifindex(i,c) for i in link_data.interfaces if i.node!=nodename ]
+       ifs.append(l)
+  return ifs
+
+def update_vlan_access_links(topology: Box, nodename: str, clones: list) -> None:
+  for vname,vdata in topology.vlans.items():                                # Iterate over global VLANs
+    if not isinstance(vdata,Box):                                           # VLAN not yet a dictionary?
+      continue                                                              # ... no problem, skip it
+    if not 'links' in vdata:                                                # No VLAN links?
+      continue                                                              # ... no problem, move on
+
+    for cnt,l in enumerate(vdata.links):                                    # So far so good, now iterate over the links
+      link_data = links.adjust_link_object(                                 # Create link data from link definition
+                    l=l,
+                    linkname=f'vlans.{vname}.links[{cnt+1}]',
+                    nodes=topology.nodes)
+      if link_data is not None:
+        ifs = clone_interfaces(link_data,nodename,clones)
+        vdata.links += ifs
+
+"""
+repeat_node - Clones a given node N times, creating additional links and/or interfaces for the new nodes
+"""
+def repeat_node(node: Box, topology: Box) -> None:
+  count = node.pop('repeat',None)
+  clones = []
+  for c in range(2,count+2):                                                # Existing node is '1'
+    clone = data.get_box(node)
+    clone.name = strings.eval_format(topology.defaults.repeater.node_name_pattern, node + { 'id': c } )
+    clone.interfaces = []                                                   # Start clean, remove reference to original node
+    if 'id' in node:
+      clone.id = node.id + c - 1                                            # Update any explicit node ID sequentially
+    topology.nodes += { clone.name: clone }
+    clones.append( clone.name )
+
+  for link in list(topology.get('links',[])):
+    link_data = links.adjust_link_object(link,f'repeater{c}',topology.nodes)
+    if link_data is None:
+      continue
+    if 'lag' in link_data:
+      log.error("LAG links not yet supported by repeater plugin, ignoring...",
+                category=Warning, module='repeater')
+      continue  
+    ifs = clone_interfaces(link_data,node.name,clones)
+    topology.links += ifs
+
+  if 'groups' in topology:
+    for group in topology.groups.values():
+      if isinstance(group,Box) and node.name in group.get('members',[]):   # Groups can include 'bool flags too
```

Comment:
I think you have to skip group names starting with "_"

Reply:
done